src/components/Checkbox/Checkbox.tsx

```diff
@@ -17,15 +19,33 @@ export interface FlowbiteCheckboxRootTheme {
 export interface CheckboxProps extends Omit<ComponentProps<'input'>, 'type' | 'ref' | 'color'> {
   theme?: DeepPartial<FlowbiteCheckboxTheme>;
   color?: keyof FlowbiteColors;
+  indeterminate?: boolean;
 }
 
 export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
-  ({ className, color = 'default', theme: customTheme = {}, ...props }, ref) => {
+  ({ className, color = 'default', theme: customTheme = {}, indeterminate, ...props }, ref) => {
     const theme = mergeDeep(getTheme().checkbox, customTheme);
 
+    const inputRef = useRef<HTMLInputElement | null>(null);
```

Comment:
can't this be done using custom data attribute prop (eg: `data-...`) instead of injecting it through ref and checking the data attribute with tailwind directly in this file? 🤔

Reply:
Hi, @SutuSebastian . I have provided my responses in this [comment](https://github.com/themesberg/flowbite-react/pull/1155#issuecomment-1896256180). Please review them at your earliest convenienceI.